ci/spelling.sh

```diff
@@ -1,7 +1,12 @@
 #!/usr/bin/env bash
 
+set -eo pipefail
+
 function spellcheck () {
-    git diff --name-only --diff-filter=AM main $HEAD | grep '\.md$' | xargs --no-run-if-empty -L1 npx cspell --show-suggestions -c ./ci/spelling-config.json
+    FAILURE=0
+    git diff --name-only --diff-filter=AM main $HEAD -- ./**/*.md ./*.md | \
+        xargs --no-run-if-empty -L1 npx cspell --show-suggestions -c ./ci/spelling-config.json || FAILURE=1
+    return $FAILURE
```

Comment:
```suggestion
    FAILURE=0
    git diff --name-only --diff-filter=AM main $HEAD | grep '\.md$' | \
        xargs --no-run-if-empty -L1 npx cspell --show-suggestions -c ./ci/spelling-config.json || FAILURE=1
    return $FAILURE
```

Something like this works with `make spelling` to pick up the markdown files correctly

Reply:
LIke mentioned, the chained pipe is problematic for picking up the errors. This would actually produce a failure if case no markdown file was edited, like in this PR. If grep doesn't find files, it returns an error, which would be picked up as a spelling error.